README.md

```diff
@@ -0,0 +1,486 @@
+
+```javascript
+//    _add :: (Number, Number) -> Number
+const _add = (x, y) => x + y;
+```
+
+The type signature indicates that `_add` takes two arguments of type `Number`,
+but this is not enforced. This allows type errors to be silently ignored:
+
+```javascript
+_add('2', '2');
+// => '22'
+```
+
+`add`, on the other hand, throws if applied to arguments of the wrong types:
+
+```javascript
+add('2', '2');
+// ! TypeError: ‘add’ expected a value of type Number as its first argument; received "2"
+```
+
+Type checking is performed as arguments are provided (rather than once all
+arguments have been provided), so type errors are reported early:
+
+```javascript
+add('1');
+// ! TypeError: ‘add’ expected a value of type Number as its first argument; received "1"
+```
+
+### Types
+
+sanctuary-def defines `Type` values for JavaScript's built-in types:
+
+  * Nullary types:
+
+    - `$.Boolean`
+    - `$.Date`
+    - `$.Error`
+    - `$.Function`
+    - `$.Null`
+    - `$.Number`
+    - `$.RegExp`
+    - `$.String`
+    - `$.Undefined`
+
+  * Unary types:
+
+    - `$.Array`
+
+sanctuary-def also provides functions for defining types.
+
+#### `TypeVariable`
+
+Polymorphism is powerful. Not being able to define a function for all types
+would be very limiting indeed: one couldn't even define the identity function!
+
+```haskell
+TypeVariable :: String -> Type
+```
+
+Before defining a polymorphic function one must first define one or more type
+variables:
+
+```javascript
+const a = $.TypeVariable('a');
+const b = $.TypeVariable('b');
+
+//    id :: a -> a
+const id = def('id', {}, [a, a], x => x);
+
+id(42);
+// => 42
+
+id(null);
+// => null
+```
+
+The same type variable may be used multiple times:
+
+```javascript
+//    cmp :: a -> a -> Number
+const cmp =
+def('cmp', {}, [a, a, $.Number], (x, y) => x < y ? -1 : x > y ? 1 : 0);
+
+cmp(42, 42);
+// => 0
+
+cmp('a', 'z');
+// => -1
+
+cmp('z', 'a');
+// => 1
+
+cmp(0, '1');
+// ! TypeError: ‘cmp’ expected a value of type Number as its second argument; received "1"
+```
+
+#### `NullaryType`
+
+`NullaryType` is used to construct types with no type variables. The `Number`
+type exported by sanctuary-def is defined via `NullaryType`, as are many of the
+other exported types.
+
+To define a nullary type `t` one must provide:
+
+  - the name of `t`; and
+
+  - a predicate which accepts any JavaScript value and returns `true` if
+    (and only if) the value is a member of `t`.
+
+```haskell
+NullaryType :: String -> (* -> Boolean) -> Type
+```
+
+For example:
+
+```javascript
+//    Integer :: Type
+const Integer = $.NullaryType(
+  'my-package/Integer',
+  x => Object.prototype.toString.call(x) === '[object Number]' &&
+       Math.floor(x) === Number(x) &&
+       x >= Number.MIN_SAFE_INTEGER &&
+       x <= Number.MAX_SAFE_INTEGER
+);
+
+//    NonZeroInteger :: Type
+const NonZeroInteger = $.NullaryType(
+  'my-package/NonZeroInteger',
+  x => Integer.test(x) && Number(x) !== 0
```

Comment:
I made these changes:

```diff
--- a/README.md
+++ b/README.md
@@ -186,10 +186,10 @@ other exported types.
 
 To define a nullary type `t` one must provide:
 
-  - the name of `t`; and
+  - the name of `t` (exposed as `t.name`); and
 
   - a predicate which accepts any JavaScript value and returns `true` if
-    (and only if) the value is a member of `t`.
+    (and only if) the value is a member of `t` (exposed as `t.test`).
 
 ```haskell
 NullaryType :: String -> (* -> Boolean) -> Type
@@ -246,15 +246,16 @@ sum(['1', '2', '3', '4']);
 
 To define a unary type `t a` one must provide:
 
-  - the name of `t`;
+  - the name of `t` (exposed as `t.name`);
 
-  - a predicate which accepts any JavaScript value and returns `true` if
-    (and only if) the value is a member of `t x` for some type `x`;
+  - a predicate which accepts any JavaScript value and returns `true`
+    if (and only if) the value is a member of `t x` for some type `x`
+    (exposed as `t.test`);
 
-  - a function which takes any value of type `t a` and returns an array of
-    the values of type `a` contained in the `t`; and
+  - a function which takes any value of type `t a` and returns an array
+    of the values of type `a` contained in the `t` (exposed as `t._1`); and
 
-  - the type of `a`.
+  - the type of `a` (exposed as `t.$1`).
 
 ```haskell
 UnaryType :: String -> (* -> Boolean) -> (t a -> [a]) -> Type -> Type
@@ -307,20 +308,21 @@ fromMaybe(0, Just('XXX'));
 
 To define a binary type `t a b` one must provide:
 
-  - the name of `t`;
+  - the name of `t` (exposed as `t.name`);
 
-  - a predicate which accepts any JavaScript value and returns `true` if
-    (and only if) the value is a member of `t x y` for some types `x` and `y`;
+  - a predicate which accepts any JavaScript value and returns `true`
+    if (and only if) the value is a member of `t x y` for some types
+    `x` and `y` (exposed as `t.test`);
 
-  - a function which takes any value of type `t a b` and returns an array of
-    the values of type `a` contained in the `t`;
+  - a function which takes any value of type `t a b` and returns an array
+    of the values of type `a` contained in the `t` (exposed as `t._1`);
 
-  - a function which takes any value of type `t a b` and returns an array of
-    the values of type `b` contained in the `t`;
+  - a function which takes any value of type `t a b` and returns an array
+    of the values of type `b` contained in the `t` (exposed as `t._2`);
 
-  - the type of `a`; and
+  - the type of `a` (exposed as `t.$1`); and
 
-  - the type of `b`.
+  - the type of `b` (exposed as `t.$2`).
 
 ```haskell
 BinaryType :: String -> (* -> Boolean) -> (t a b -> [a]) -> (t a b -> [b]) -> Type -> Type -> Type
```

Thanks for catching this omission. It's good to be documenting `_1`, `_2`, `$1`, and `$2` as well!